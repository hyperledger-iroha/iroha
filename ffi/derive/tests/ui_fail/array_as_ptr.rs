use iroha_ffi::ffi_export;

/// Array as argument
#[ffi_export]
pub fn array_arg(_arr: [u32; 2]) {}

fn main() {
    let arg = [12_u32, 42_u32];
    let mut store = Default::default();
    unsafe { __array_arg(arg) };
}
