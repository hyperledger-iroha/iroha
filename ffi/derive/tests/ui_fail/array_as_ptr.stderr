error[E0308]: mismatched types
 --> tests/ui_fail/array_as_ptr.rs:9:17
  |
9 |     __array_arg(arg);
  |     ----------- ^^^ expected `*mut [u32; 2]`, found `[u32; 2]`
  |     |
  |     arguments to this function are incorrect
  |
  = note: expected raw pointer `*mut [u32; 2]`
                   found array `[u32; 2]`
note: function defined here
 --> tests/ui_fail/array_as_ptr.rs:4:1
  |
4 | #[ffi_export]
  | ^^^^^^^^^^^^^
  = note: this error originates in the attribute macro `ffi_export` (in Nightly builds, run with -Z macro-backtrace for more info)
